cli/src/main/java/com/devonfw/tools/ide/tool/ide/IdeaPluginDownloader.java

```diff
@@ -131,24 +138,34 @@ private void extractDownloadedPlugin(FileAccess fileAccess, Path downloadedFile,
   }
 
   private String getFileExtensionFromUrl(String urlString) throws IOException {
-    URL url = new URL(urlString);
-    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-    connection.setRequestMethod("HEAD");
-    connection.connect();
-
-    int responseCode = connection.getResponseCode();
-    if (responseCode != HttpURLConnection.HTTP_OK) {
-      throw new IOException("Failed to fetch file headers: HTTP " + responseCode);
-    }
 
-    String contentType = connection.getContentType();
-    if (contentType == null) {
-      return "";
+    URI uri = null;
+    HttpRequest request;
+    try {
+      uri = URI.create(urlString);
+      request = HttpRequest.newBuilder().uri(uri)
+          .method("HEAD", HttpRequest.BodyPublishers.noBody()).timeout(Duration.ofSeconds(5)).build();
+
+      HttpResponse<?> res = this.client.send(request, HttpResponse.BodyHandlers.ofString());
+
+      int responseCode = res.statusCode();
+      // TODO: brauchen wir die if-abfrage überhaupt noch, oder wird der Fehler durch catch-block behandelt
+      if (responseCode != HttpURLConnection.HTTP_OK) {
+        throw new RuntimeException("Failed to fetch file headers: HTTP " + responseCode);
+      }
+
+      String contentType = res.headers().firstValue("content-type").orElse("undefined");
+      if (contentType.equals("undefined")) {
+        return "";
+      }
+      return switch (contentType) {
```

Comment:
IMHO the real value of `Optional` in Java is questionable.
Mainly it exists because developers are not able to handle `null` properly and are too lazy for `null` checks.
Doing `String value = optional.orElse("«fallback»")` and then doing `if (value.equals("«fallback»") { ... }` is not a great idea.
In such case better do something like this:
```suggestion
      Optional<String> contentType = res.headers().firstValue("content-type")
      if (contentType.isPresent()) {
        return "";
      }
      return switch (contentType.get()) {
```